### PR TITLE
KTOR-6156 docs for CallId client plugin

### PR DESCRIPTION
**Pull request by vnikolova**
[KTOR-6156](https://youtrack.jetbrains.com/issue/KTOR-6156/Documentation-for-CallId-Feature-for-Client)
- added documentation for CallId client plugin
- created a new code snippet with example. The example uses a Ktor server to demonstrate how call ID is transferred between multiple calls
- added a link summary to the CallId server plugin doc

### CODE REVIEW

File: codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import io.ktor.client.statement.*
+import io.ktor.client.plugins.callid.*
+import io.ktor.http.*
+import kotlinx.coroutines.*
+
+fun main() {
+    runBlocking {
+        val client = HttpClient(CIO) {
+            install(CallId) {
+                useCoroutineContext = true
```
Comment (rsinukov) [inline codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt:15]: I think we should not explicitly set default values in config

File: codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/CallIdService.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import io.ktor.server.routing.*
+
+fun main(args: Array<String>) {
+    embeddedServer(Netty, port = 8090, host = "0.0.0.0", module = Application::module)
+        .start(wait = true)
+}
+
+fun Application.module() {
+    val client = HttpClient(CIO) {
+        install(ClientCallId) {
```
Comment (rsinukov) [inline codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/CallIdService.kt:23]: unnecessary `{}`

File: topics/client_call_id.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+The %plugin_name% plugin configuration, provided by the [CallIdConfig]() class, allows you to generate a call ID and add
+it to the call context.
+
+### Generate a call ID
+
+Generate a call ID for a specific request in one of the following ways:
+
+* The `useCoroutineContext` property, enabled by default, adds a generator that uses the current `CoroutineContext` to
+  retrieve a call ID.
```
Comment (rsinukov) [inline topics/client_call_id.md:44]: It worth mentioning that server plugin adds call id to the coroutine context

**Review by rsinukov — APPROVED** [review]

File: codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt (hunk 1 above)
Comment (vnikolova) [reply codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt]: I set it to false in the next example and adjusted the docs.

File: codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import io.ktor.client.engine.cio.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.client.plugins.callid.*
+import io.ktor.http.*
+import kotlinx.coroutines.*
+
+fun main() {
+    runBlocking {
+        val client = HttpClient(CIO) {
```
Comment (e5l) [inline codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt:13]: I would extract every plugin usage to a separate function with a specific name

File: codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            }
+        }
+
+        val response: HttpResponse = client.get("http://0.0.0.0:8090/call")
+        println(response.bodyAsText())
+
+        val responseFromDoubleCall: HttpResponse = client.get("http://0.0.0.0:8090/double-call")
+        println(responseFromDoubleCall.bodyAsText())
+
+        val client2 = HttpClient(CIO) {
```
Comment (e5l) [inline codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt:27]: client2 should also be closed.

I would write it using `use` extension: 
```
client2.use {
}
```

File: codeSnippets/snippets/client-call-id/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+```bash
+./gradlew :client-call-id:run
+```
+
+The client makes 3 requests to the API and prints out information about the call ID.
+
+The expected output is:
+```
+in call: call-id-client, in header: call-id-client, in query param: call-id-client
```
Comment (e5l) [inline codeSnippets/snippets/client-call-id/README.md:24]: Could we print it in a way that indicates how the data flow goes? For instance, in a server client we can use call id from the request as a part, so we can track the whole chain of the calls

**Review by e5l — APPROVED** [review]
LGTM in general. We can improve samples a bit

File: codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt (hunk 4 above)
Comment (vnikolova) [reply codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt:13]: Do you mean like this:

```kotlin
fun main() {
        getRequestWithCustomCallId()
        //...
}

fun getRequestWithCustomCallId() {
    runBlocking {
        val client = HttpClient(CIO) {
            install(CallId) {
                generate { "call-id-client" }
                addToHeader(HttpHeaders.XRequestId)
                intercept { request, callId -> request.parameter("callId", callId) }
            }
        }
        val response: HttpResponse = client.get("http://0.0.0.0:8090/call")
        println(response.bodyAsText())

        val responseFromDoubleCall: HttpResponse = client.get("http://0.0.0.0:8090/double-call")
        println(responseFromDoubleCall.bodyAsText())
        client.use {
            client.close()
        }
    }
}
//...
```

Comment (e5l) [reply codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt:13]: fun Closeable.use(block: () -> Unit) {
}

**Review by e5l — APPROVED** [review]

**Pull request merged**